cpp/lib/ome/xml/meta/Convert.cpp

```diff
@@ -229,6 +228,43 @@ namespace
       return ok;
     }
 
+    /**
+     * Transfer a single metadata value (four index arguments).
+     *
+     * @param get a getter method from MetadataRetrieve.
+     * @param set a setter method from MetadataStore.
+     * @returns @c true if the transfer succeeded, @c false if an
+     * exception was thrown.
+     */
+    template<typename T, typename P>
+    bool
+    transfer(T    (MetadataRetrieve::* get)(P param1,
+                                            P param2,
+                                            P param3,
+                                            P param4) const,
+             void (MetadataStore::*    set)(T value,
+                                            P param1,
+                                            P param2,
+                                            P param3,
+                                            P param4),
+             P                         param1,
+             P                         param2,
+             P                         param3,
+             P                         param4)
+    {
+      bool ok = true;
+      try
+        {
+          (dest.*set)((src.*get)(param1, param2, param3, param4),
+                      param1, param2, param3, param4);
+        }
+      catch (const std::runtime_error& /* e */)
```

Comment:
I didn't follow up to this one.  I don't really like this, but it's matching the behaviour of the Java implementation.  It also means that we can parse a document with minor errors.  It will need logging adding to log what the error was, once we've added a logging library (this is on the todo list).